aria_data_loaders/aria_data_utils/episodic_memory_robotic_fetch.py

```diff
@@ -0,0 +1,474 @@
+import rospy
+import sophus as sp
+from aria_data_utils.conversions import (
+    ros_PoseStamped_to_sp_SE3,
+    ros_TransformStamped_to_sophus_SE3,
+    sophus_SE3_to_ros_PoseStamped,
+    sophus_SE3_to_ros_TransformStamped,
+    xyt_to_sophus_SE3,
+)
+from geometry_msgs.msg import PoseStamped
+from spot_rl.envs.skill_manager import SpotSkillManager
+from spot_rl.utils.utils import ros_frames as rf
+from spot_wrapper.spot import Spot
+from spot_wrapper.spot_qr_detector import SpotQRDetector
+from std_msgs.msg import Bool
+from tf2_ros import (
+    ConnectivityException,
+    ExtrapolationException,
+    LookupException,
+    StaticTransformBroadcaster,
+)
+from tf2_ros.buffer import Buffer
+from tf2_ros.transform_listener import TransformListener
+
+# For information on useful reference frame, please check https://github.com/facebookresearch/spot-sim2real?tab=readme-ov-file#eyeglasses-run-spot-aria-project-code
+
+# Waypoints on left of the dock to observe the QR code clearly
+SPOT_DOCK_OBSERVER_WAYPOINT_LEFT = [
+    0.3840912007597151,
+    -0.5816728569741766,
+    149.58030524832756,
+]
+# Waypoints on right of the dock to observe the QR code clearly
+SPOT_DOCK_OBSERVER_WAYPOINT_RIGHT = [
+    0.5419185005639034,
+    0.5319243891865243,
+    -154.1506754378722,
+]
+
+
+class EpisodicMemoryRoboticFetch:
+    """
+    This class handles the core logic of the Episodic Memory Robotic Fetch
+    It begines with Spot looking at the marker and detecting it which results
+    in a static tf of marker w.r.t spotWorld frame being published.
+
+    Then it waits for Aria to detect marker and publish ariaWorld w.r.t marker
+    as a static tf. This is done by AriaLiveReader node. This step ensures both
+    Spot & Aria worlds are now mutually transformable which is necessary for the
+    rest of the logic.
+
+    It constantly listens to Aria's current pose w.r.t ariaWorld frame.
+    It also waits for Aria to detect the object and publish that pose of interest
+    (i.e. aria's cpf frame when it saw the object) w.r.t ariaWorld frame.
+
+    Once both the poses are received, it waits for the spot_fetch trigger to be True
+    which will be set by the user via `rostopic pub` on cli.
+
+    Once the trigger is True, it will run the following steps:
+        1. Nav to Pose of interest (Aria's cpf frame when it saw the object)
+        2. Pick the object (object is hardcoded to be bottle)
+        3. Nav to Pose of interest (Aria wearer's last location before triggering spot_fetch)
+        4. Place the object (object is hardcoded to be bottle)
+
+    Args:
+        spot: Spot object
+        verbose: bool indicating whether to print debug logs
+
+    Warning: Do not create objects of this class directly. It is a standalone node
+    """
+
+    def __init__(self, spot: Spot, verbose: bool = True, use_policies: bool = True):
+        # Get Spot Skill Manager
+        # @FIXME: This is initializing a ros-node silently (at core of inheritence in
```

Comment:
should we add this to our next PR, where we clean up functions and move them to utils etc?

Reply:
For this particular FIXME, we have an issue up (https://github.com/facebookresearch/spot-sim2real/issues/115).
This one in particular is more involved issue and might need deeper changes in code.

But I agree about other cleanups. We have started working on the migration and cleanup of utils to a separate package in spot-sim2real @rutadesai